dso-l2/src/main/java/com/tc/objectserver/entity/ClientDescriptorImpl.java

```diff
@@ -64,13 +63,9 @@ public int hashCode() {
   
   @Override
   public boolean equals(Object other) {
-    boolean doesMatch = (this == other);
-    if (!doesMatch && (getClass() == other.getClass()))
-    {
-      final ClientDescriptorImpl that = (ClientDescriptorImpl) other;
-      doesMatch = this.nodeID.equals(that.nodeID)
-          && this.entityDescriptor.equals(that.entityDescriptor);
-    }
-    return doesMatch;
+    if (this == other) return true;
+    if (other == null || getClass() != other.getClass()) return false;
+    final ClientDescriptorImpl that = (ClientDescriptorImpl) other;
```

Comment:
ok! 
You mean it would be better to cast against the interface ?
Let me know if you need to test you changes :-)